client/task-list/tasks/payments/PaymentGatewaySuggestions/components/List/test/list.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { List } from '../List';
+
+jest.mock( '@woocommerce/tracks', () => ( {
+	recordEvent: jest.fn(),
+} ) );
+
+jest.mock( '@wordpress/data', () => {
+	// Require the original module to not be mocked...
+	const originalModule = jest.requireActual( '@wordpress/data' );
+
+	return {
+		__esModule: true, // Use it when dealing with esModules
+		...originalModule,
+		useSelect: jest.fn().mockReturnValue( {
+			menuItems: [],
+		} ),
+	};
+} );
+
+const mockGateway = {
+	id: 'mock-gateway',
+	title: 'Mock Gateway',
+	plugins: [],
+	postInstallScripts: [],
+	requiredSettings: [],
+	installed: false,
+	needsSetup: false,
+	enabled: false,
+};
+
+const defaultProps = {
+	heading: 'Test heading',
+	markConfigured: jest.fn(),
+	recommendation: 'testId',
+	paymentGateways: [ mockGateway ],
+};
+
+describe( 'PaymentGatewaySuggestions > List', () => {
+	it( 'should display correct heading', () => {
+		const { queryByText } = render( <List { ...defaultProps } /> );
+
+		expect( queryByText( defaultProps.heading ) ).toBeInTheDocument();
+	} );
+
+	it( 'should display gateway title', () => {
+		const { queryByText } = render( <List { ...defaultProps } /> );
+
+		expect( queryByText( mockGateway.title ) ).toBeInTheDocument();
+	} );
+
+	it( 'should display the "Enable" button when setup is NOT required', () => {
+		const { queryByRole } = render( <List { ...defaultProps } /> );
+
+		expect( queryByRole( 'button' ) ).toHaveTextContent( 'Enable' );
+	} );
+
+	it( 'should display the "Set up" button when setup is required', () => {
+		const props = {
+			...defaultProps,
+			paymentGateways: [
+				{
+					...mockGateway,
+					needsSetup: true,
+					plugins: [ 'test-plugins' ],
+				},
+			],
+		};
+
+		const { queryByRole } = render( <List { ...props } /> );
+
+		expect( queryByRole( 'button' ) ).toHaveTextContent( 'Set up' );
+	} );
+
+	it( 'should display the SetupRequired component when appropriate', () => {
+		const props = {
+			...defaultProps,
+			paymentGateways: [
+				{
+					...mockGateway,
+					needsSetup: true,
+					installed: true,
+					plugins: [ 'test-plugin' ],
+				},
+			],
+		};
+
+		const { queryByText } = render( <List { ...props } /> );
+
+		expect( queryByText( 'Setup required' ) ).toBeInTheDocument();
+	} );
+
+	it( 'should display the Recommended ribbon when appropriate', () => {
```

Comment:
Should we also include a negative test case to make sure this isn't always shown?

Reply:
Good call, added in a0baa12dfc1c7c6dca1e042e89396ee11395f8e7